### PR TITLE
Fixed stacktraces caused by incorrect JNDI lookup

### PR DESCRIPTION
**Pull request by dmatej**
* Fixed processing default local interface
  - cause:
    - the stateless bean implemented some interface from library
    - the interface was annotated only with `@Path`
    - neither interface nor class was annotated by `@Local` or `@Remote` (then `@Local` is implicit in ejb3.2)

* Implemented support of `Stateless.name` attribute
  - it is used in JNDI names on Payara

* Limited default local interfaces
  - partial impl of the ejb 3.2 spec 4.9.7, some interfaces are not allowed

Signed-off-by: David Matejcek <dmatej@seznam.cz>

### CODE REVIEW

**Comment by dmatej** [comment]
Is the failing test stable?

**Comment by jansupol** [comment]
This test is intermittently failing, sorry for that, for all JDKs. Should be fixed by #4387.

File: containers/glassfish/jersey-gf-ejb/src/main/java/org/glassfish/jersey/gf/ejb/internal/EjbComponentProvider.java [hunk 1]
```diff
@@ -92,23 +95,43 @@
...
-                return (T) lookup(ctx, clazz, clazz.getSimpleName(), ejbProvider);
+                return (T) lookup(ctx, clazz, beanName, ejbProvider);
             } catch (NamingException ex) {
                 Logger.getLogger(ApplicationHandler.class.getName()).log(Level.SEVERE, null, ex);
                 return null;
             }
         }
 
+        private static <T> String getBeanName(final Class<T> clazz) {
+            final Stateless stateless = clazz.getAnnotation(Stateless.class);
```
Comment (jansupol) [inline containers/glassfish/jersey-gf-ejb/src/main/java/org/glassfish/jersey/gf/ejb/internal/EjbComponentProvider.java]: Why don't you deal with `Stateful`, too?

Comment (dmatej) [reply containers/glassfish/jersey-gf-ejb/src/main/java/org/glassfish/jersey/gf/ejb/internal/EjbComponentProvider.java]: It is not requested by specification. Isn't REST/Jersey stateless?

Comment (jansupol) [reply containers/glassfish/jersey-gf-ejb/src/main/java/org/glassfish/jersey/gf/ejb/internal/EjbComponentProvider.java]: You are right, JAX-RS Spec discusses Stateless and Singleton, only.

**Pull request merged**